server.js

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable no-console */
+
+const assert = require('assert');
+const camelCase = require('camelcase');
+
+const Provider = require('oidc-provider');
+
+const port = process.env.PORT || 3000;
+
+const config = ['CLIENT_ID', 'CLIENT_SECRET', 'CLIENT_REDIRECT_URI'].reduce((acc, v) => {
+  assert(process.env[v], `${v} config missing`);
+  acc[camelCase(v)] = process.env[v];
+  return acc;
+}, {});
+
+const oidcConfig = {
+  features: {
+    devInteractions: true,
+    discovery: true,
+    registration: false,
+    revocation: true,
+    sessionManagement: false
+  },
+  format: {
+    default: 'jwt',
+    AccessToken: 'jwt',
+    RefreshToken: 'jwt'
+  }
+};
+
+const oidc = new Provider(`http://localhost:${port}`, oidcConfig);
```

Comment:
I assume the host doesn't need to be configurable?

Reply:
Initially, I wanted to make this configurable. Then figured that hardcoding it to `localhost` would emphasise the fact that this is purely for local development. Can always make it configurable later if the host needs to be customisable (though at the moment I can't see a scenario where it would be needed).